src/hotspot/share/opto/compile.cpp

```diff
@@ -3802,6 +3798,34 @@ void Compile::final_graph_reshaping_main_switch(Node* n, Final_Reshape_Counts& f
   }
 }
 
+void Compile::remove_range_check_cast(CastIINode* cast) {
```

Comment:
Why not make this a member function of `CastIINode`?

Reply:
Done in new commit.